src/test/java/spoon/test/template/TemplateTest.java

```diff
@@ -107,7 +107,8 @@ public void testTemplateC1() throws Exception {
 		assertEquals("return \"testparam2\"", m2.getBody().getStatement(0)
 				.toString());
 
-		new ModelConsistencyChecker(factory.getEnvironment(), false, true).scan(c1);
+		// fixInconsistencies is set to true because references don't have a null parent.
```

Comment:
Yes, they have a parent. Before the application of the scanner `ModelConsistencyChecker`, the parent of references are elements of the template. The scanner set the parent after the substitution of templates and put the target parent in references.